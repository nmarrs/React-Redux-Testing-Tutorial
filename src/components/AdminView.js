import React from 'react'
import PropTypes from 'prop-types'

import TaskColumn from './TaskColumn'

const AdminView = (props) => {
  const {
    templateTasks,
    toggleEditTaskModal,
    editTaskModalOpen,
    isAdminView,
    editTask,
    currentTask
  } = props

  return (
    <div>
      <h3 className='mt-3'><i className="fas fa-tasks"></i> Template Tasks</h3>
      <TaskColumn
        title={'Tasks'}
        tasks={templateTasks}
        toggleEditTaskModal={toggleEditTaskModal}
        editTaskModalOpen={editTaskModalOpen}
        isAdminView={isAdminView}
        editTask={editTask}
        currentTask={currentTask} />
    </div>
  )
}

AdminView.propTypes = {
  templateTasks: PropTypes.array.isRequired,
  isAdminView: PropTypes.bool.isRequired,
  editTaskModalOpen: PropTypes.bool.isRequired,
  toggleEditTaskModal: PropTypes.func.isRequired,
  currentTask: PropTypes.object
}

export default AdminView
