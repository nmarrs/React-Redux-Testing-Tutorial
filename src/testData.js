export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
