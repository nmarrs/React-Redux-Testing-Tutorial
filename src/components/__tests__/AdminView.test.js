import React from 'react'
import { shallow } from 'enzyme'

import AdminView from '.././AdminView'
import { testTask, testTask1 } from '../../testData'

describe('AdminView', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<AdminView
      templateTasks={[ testTask, testTask1 ]}
      isAdminView={true}
      editTaskModalOpen={false}
      toggleEditTaskModal={jest.fn()} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
