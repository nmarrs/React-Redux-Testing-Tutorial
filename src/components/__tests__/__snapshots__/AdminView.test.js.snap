// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminView renders correctly 1`] = `
<div>
  <h3
    className="mt-3"
  >
    <i
      className="fas fa-tasks"
    />
     Template Tasks
  </h3>
  <TaskColumn
    editTaskModalOpen={false}
    isAdminView={true}
    tasks={
      Array [
        Object {
          "completionTime": "",
          "currentStatus": "To Do",
          "endTime": "",
          "feedback": "",
          "id": 1,
          "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
          "priority": 3,
          "rankWeight": 150,
          "startTime": "",
          "timeEstimate": 2,
          "title": "Unload merchandise",
        },
        Object {
          "completionTime": "",
          "currentStatus": "To Do",
          "endTime": "",
          "feedback": "",
          "id": 2,
          "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
          "priority": 1,
          "rankWeight": 10,
          "startTime": "",
          "timeEstimate": 10,
          "title": "Stock shelves with unpacked items",
        },
      ]
    }
    title="Tasks"
    toggleEditTaskModal={[MockFunction]}
  />
</div>
`;
